src/damavand/cloud/aws/resources/vllm_component.py

```diff
@@ -359,41 +335,150 @@ def api_resource_completions(self) -> aws.apigateway.Resource:
             path_part="completions",
         )
 
+
     @property
     @cache
     def api_method(self) -> aws.apigateway.Method:
         """
         Return a method for the API Gateway.
+        """
+
+        if self.args.public_internet_access:
+            return aws.apigateway.Method(
+                resource_name=f"{self._name}-api-method",
+                opts=ResourceOptions(parent=self),
+                rest_api=self.api.id,
+                resource_id=self.api_resource_completions.id,
+                http_method="POST",
+                authorization="NONE",
+            )
+        else:
+            return aws.apigateway.Method(
+                resource_name=f"{self._name}-api-method",
+                opts=ResourceOptions(parent=self),
+                rest_api=self.api.id,
+                resource_id=self.api_resource_completions.id,
+                http_method="POST",
+                authorization="NONE",
+                api_key_required=True,
+            )
```

Comment:
You can remove the if condition just by passing the `public_internet_access` to the `api_key_required` argument:

```suggestion
        return aws.apigateway.Method(
            resource_name=f"{self._name}-api-method",
            opts=ResourceOptions(parent=self),
            rest_api=self.api.id,
            resource_id=self.api_resource_completions.id,
            http_method="POST",
            authorization="NONE",
            api_key_required=self.args.public_internet_access,
        )
```